Add test for Web MIDI API.

ryoyakawai opened the pull request:
Referred [Web MIDI API Editor's Draft](http://webaudio.github.io/web-midi-api/).
Web MIDI API is only implemented in Chrome (as is 2015/02/20).

<!-- Reviewable:start -->

<!-- Reviewable:end -->

ryoyakawai committed:
initial import of Web MIDI API Test

ryoyakawai committed:
Modified comment format.

sideshowbarker commented:
We need to identify a good reviewer for this PR. Anybody have suggestions?

myakura commented:
perhaps people on https://github.com/WebAudio/web-midi-api/graphs/contributors could help? +@ryoyakawai

KenjiBaheux commented:
@cwilso @sideshowbarker what do you think about the suggestion in https://github.com/w3c/web-platform-tests/pull/1629#issuecomment-76957128 ?

sideshowbarker commented:
> @cwilso @sideshowbarker what do you think about the suggestion in #1629 (comment) ?

Agreed it would be great if @jussi-kalliokoski or @toyoshim could step in to help out with review on this if they have time. I’d be glad to help them with getting familiar with testharness.js if needed—or glad to help in any other way I can with facilitating the review.

ryoyakawai commented:
@myakura , @KenjiBaheux , @sideshowbarker  Thank you very much!

@cwilso , @jussi-kalliokoski , @toyoshim I appreciate if you review the test and leave comments.

toyoshim commented:
I can take a look from Web MIDI spec side, but I'm not familiar with web platform tests and Critic.
But anyway, I'll try doing my best.
It would be nice if someone also review from viewpoints of test frameworks.

ryoyakawai committed:
https://critic.hoppipolla.co.uk/showcomment?chain=11098
Deleted .gitignore from this directory.

ryoyakawai committed:
Modified.
https://critic.hoppipolla.co.uk/showcomment?chain=11099

ryoyakawai committed:
Replaces assert_false(true) to assert_unreached()
https://critic.hoppipolla.co.uk/showcomment?chain=11101

ryoyakawai committed:
Changed assert_true() to assert_equals() here and there.
https://critic.hoppipolla.co.uk/showcomment?chain=11103

ryoyakawai committed:
Updated assert_own_property() to assert_idl_attribute(). And added more property test to follow wd.
https://critic.hoppipolla.co.uk/showcomment?chain=11103

ryoyakawai committed:
changed to use assert_class_string() instead of hoge.constructor.name.
https://critic.hoppipolla.co.uk/showcomment?chain=11102

updated tests to match wd.
https://critic.hoppipolla.co.uk/showcomment?chain=11104

ryoyakawai committed:
Modified test on errorCallback.
https://critic.hoppipolla.co.uk/showcomment?chain=11106

ryoyakawai committed:
Added TODO/FIXME comment for the MIDIOptions flag

toyoshim commented:
I starts review cycles. But since this is my first challenge to use OperaCritic and review tests for web-platform-tests, please let me know if I make mistakes.

ryoyakawai committed:
Updated valuable name from tmp to outputEntries and outputEntries.
https://critic.hoppipolla.co.uk/showcomment?chain=11109

ryoyakawai committed:
Updated: update against spec changes
https://critic.hoppipolla.co.uk/showcomment?chain=11104

ryoyakawai committed:
Delete: WhiteSpace

ryoyakawai committed:
Delete: console.log()

ryoyakawai committed:
Delete: Completely deleted console.log(hoge);

ryoyakawai committed:
Modify: change commented script to alive script
https://critic.hoppipolla.co.uk/showcomment?chain=11110

ryoyakawai committed:
Update: Comment Style
https://critic.hoppipolla.co.uk/showcomment?chain=11131

ryoyakawai committed:
Modify: change assert_equals to assert_class_string.
https://critic.hoppipolla.co.uk/showcomment?chain=11138

ryoyakawai committed:
Modify: "assert_equals" to "assert_class_string"
https://critic.hoppipolla.co.uk/showcomment?chain=11141

ryoyakawai committed:
Modify: assert_equals to assert_any.
https://critic.hoppipolla.co.uk/showcomment?chain=11143

ryoyakawai committed:
Fixed TODO style
https://critic.hoppipolla.co.uk/e0f3a468?review=4026

ryoyakawai committed:
Delete: error state "AbortError", "InvalidStateError"
https://critic.hoppipolla.co.uk/showcomment?chain=11143

ryoyakawai committed:
Modified: devide for 2 tests: - sysex: true, sysex: false
https://critic.hoppipolla.co.uk/showcomment?chain=11144

ryoyakawai committed:
Fix: put attribute for forEach(), change "Undefined" from "undefined"

ryoyakawai committed:
Add: TODO

ryoyakawai committed:
Delete: White Space

toyoshim commented:
We finished a review, but could not figure out how one test can be marked as a manual test. Anyone can help this? Once the problem is fixed, this suite looks good to me.

sideshowbarker commented:
> We finished a review, but could not figure out how one test can be marked as a manual test.

The normal way to indicate to the harness and test runner that a test is a manual test is just to name it with `-manual.html` suffix instead of just `.html`. Do you mean something other than that?

sideshowbarker commented:
@toyoshim https://critic.hoppipolla.co.uk/2a218925?review=4026 has not been reviewed yet, as far as the critic system is concerned. Can you please go there and (re)review it and mark it as reviewed in critic?

ryoyakawai committed:
Replace automatic test with manual test
 Delete: automatic test for MIDI Input/Output
 Add: Manual test for MIDI Input/Outpu

ryoyakawai committed:
Delete: white space
